ui/src/components/events/EventCard.js

```diff
@@ -117,33 +117,44 @@ const EventCard = (props) => {
               Location: {eventData.location}
             </Text>
           )}
-          <Box fontSize={["xs", "md"]} mt="5" noOfLines="4">
+          <Text fontSize={["xs", "md"]} mt="5">
             {parse(eventData.description, options)}
-          </Box>
+          </Text>
+          <Box
+            position="absolute"
+            bottom="0"
+            left="0"
+            w="100%"
+            h="25%"
+            background="linear-gradient(to bottom,  rgba(255,255,255,0) 0%,rgba(255,255,255,1) 100%); "
+          ></Box>
         </Box>
         <Stack mt="5" direction="row" justifyContent="center" spacing={2}>
-          <Button
-            flex={1}
-            as={Link}
-            size="sm"
-            target="_blank"
-            href={eventData.signUpLink ? eventData.signUpLink : false}
-            isDisabled={eventData.signUpLink.length <= 0}
-            fontSize={["xs", "sm"]}
-          >
-            Sign up
-          </Button>
-          <Button
-            flex={1}
-            as={Link}
-            size="sm"
-            target="_blank"
-            href={generateGoogleCalendarLink(eventData)}
-            isDisabled={generateGoogleCalendarLink(eventData) ? false : true}
-            fontSize={["xs", "sm"]}
-          >
-            Add to Calendar
-          </Button>
+          {eventData.signUpLink.length > 0 && (
+            <Button
+              flex={1}
+              as={Link}
+              size="sm"
+              target="_blank"
+              href={eventData.signUpLink ? eventData.signUpLink : false}
```

Comment:
this returns error. you cannot pass a boolean to href. return null instead. same for your other buttons and your generateGoogleCalendarLink

Reply:
also, i'll prob do it in my homepage branch, but i think a lot of this logic should be in our backend

Reply:
yea, if it is available directly from our backend it makes a lot of this so much easier and for frontend it should just be getting data from BE. temporary workaround for now since that involves editing the advanced custom fields